src/linux/ebpf.cpp

```diff
@@ -151,6 +163,52 @@ Try<Nothing> attach(int fd, const string& cgroup)
   return Nothing();
 }
 
+
+Try<vector<uint32_t>> attached(const string& cgroup)
+{
+  Try<int> cgroup_fd = os::open(cgroup, O_DIRECTORY | O_RDONLY | O_CLOEXEC);
+  if (cgroup_fd.isError()) {
+    return Error("Failed to open '" + cgroup + "': " + cgroup_fd.error());
+  }
+
+  // Program ids are unsigned 32-bit integers. We assume that a maximum
+  // of 32 programs are attached to a cgroup; there should only be 0 or 1 but
+  // we allow for more to be safe. We initialize our query buffer with -1 so
+  // we can distinguish between a returned program id and filler values.
+  //
+  // Example: [1, 7, 3, -1, -1, -1, ..., -1] => [1, 7, 3] are program ids.
+  const int MAX_IDS = 32;
+  vector<int32_t> buffer(MAX_IDS, -1);
```

Comment:
looks like the output size is from attr.query.prog_cnt, strangely enough, we could simplify things here if we use that rather than inferring from the values of the ids